### PR TITLE
Add create command to cli

### PR DESCRIPTION
**Pull request by andreas-marschke**
Adds a `create` command to the commandline. Takes a JSON file as input describing the type of object and it's attributes.

Please review: @msolnit @nicjansma

### CODE REVIEW

File: cli/create.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            var filestr = fs.readFileSync(filePath, 'utf8');
+            var data;
+
+            try {
+                data = JSON.parse(filestr);
+            }
+            catch(err) {
+                cmdCore.handleError(err);
+            }
+						log.debug(data);
```
Comment (msolnit) [inline cli/create.js]: Some weird whitespace on this line.

Comment (andreas-marschke) [inline cli/create.js]: Ugh shucks sorry. Editor was in boomerang mode :(
Am 15.10.2015 18:07 schrieb "Matt Solnit" <notifications@github.com>:

**Comment by msolnit** [comment]
@andreas-marschke other than the indentation comment, this looks good to me.  Two ideas for enhancement:
* support for reading from stdin instead of a file
* support for JSON output, to make it machine-readable for scripting (e.g. instead of `New ID is: 42` you could opt for `{"id": 42}`

**Comment by nicjansma** [comment]
:+1: to Matt's thoughts.

Otherwise looks good.

**Comment by andreas-marschke** [comment]
@msolnit Updated

File: cli/cmd.js [hunk 2]
```diff
@@ -33,8 +34,12 @@ program.command("delete <type> <id>")
     .description("delete objects")
     .action(require("./delete.js"));
 
-program.command("update <type> <id> <file>")
+program.command("update <type> <id> [file]")
     .description("Update an object in the repository based on the JSON formatted data in a file")
     .action(require("./update.js"));
 
+program.command("create [file]")
+    .description("Create a new object of type <type> based on the JSON data in <file>")
```
Comment (msolnit) [inline cli/cmd.js]: @andreas-marschke there's no `type` parameter; the type is simply based on whatever's in the JSON input.

Comment (andreas-marschke) [inline cli/cmd.js]: Fixed.

File: cli/update.js [hunk 3]
```diff
@@ -13,18 +13,50 @@ var path = require("path");
...
+        var fullPath = path.resolve(file);
+        var jsonObject;
+
+        try {
+            fs.existsSync(fullPath);
+            jsonObject = require(fullPath);
+        }
+        catch(exception) {
+            cmdCore.handleError(exception);
+            process.exit(1);
```
Comment (msolnit) [inline cli/update.js]: @andreas-marschke `handleError()` already calls `process.exit(1)`, right?

Comment (andreas-marschke) [inline cli/update.js]: @msolnit Good Catch!

**Comment by msolnit** [comment]
OK, works for me!

**Pull request merged**